electrumsv/wallet_database/functions.py

```diff
@@ -1090,27 +1090,31 @@ def read_payment_request_outputs(db: sqlite3.Connection, paymentrequest_ids: lis
     return read_rows_by_id(PaymentRequestOutputRow, db, sql, (), paymentrequest_ids)
 
 
-def create_pushdata_matches_write(rows: list[PushDataMatchRow], processed_message_ids: list[int],
+def create_pushdata_match_write(row: PushDataMatchRow, processed_message_ids: list[int],
         db: Optional[sqlite3.Connection]=None) -> None:
     assert db is not None and isinstance(db, sqlite3.Connection)
     sql = """
     INSERT INTO ServerPushDataMatches (server_id, pushdata_hash, transaction_hash,
         transaction_index, block_hash, match_flags, date_created)
     VALUES (?,?,?,?,?,?,?)
     """
-    db.executemany(sql, rows)
-
-    if len(processed_message_ids) > 0:
-        update_server_peer_channel_message_flags_write(processed_message_ids, db)
+    try:
```

Comment:
This entire change could be reduced to:
1. Given the original form of this function.
2. Loop over the rows doing a singlar execute for each.
3. Swallow any integrity error and log it in debug level.
4. Add a comment why we are swallowing the integrity error "tip filter used to detect new transactions featuring the pushdata.."

Reply:
Okay. Sure. The rationale to raise and catch it in the `wallet.py` was to increase transparency about what is going on there if the code is re-visited.

Reply:
So then that comment in the header of the function should be that we ignore tip filter events for the same pushdata/tx in the database function, instead of constructing code to send a message?